src/lib/utils/envHelper.ts

```diff
@@ -116,27 +118,41 @@ const fillNetworkData = (network: NetworkDataFromEnvironmentType, networkId: num
 
 export const NETWORKS_DATA = data ? JSON.parse(data).map((network: NetworkDataFromEnvironmentType) => fillNetworkData(network, network.networkId, network.apiVersion)) : null;
 
+export const getCurrentNetwork = () => {
+  if (!NETWORKS_DATA && !process.env.REACT_APP_DISABLE_NETWORK_DATA_CHECK) throw new Error('Provide network data');
+  const currentNetworkIndex = window.localStorage.getItem('current_network');
+  if (currentNetworkIndex && NETWORKS_DATA.length > 1) {
+    const networkId = NETWORKS_DATA[currentNetworkIndex].networkId;
+    return { ...NETWORKS_DATA[currentNetworkIndex], ...NETWORK_VARIABLES[networkId] };
+  }
+  if (!currentNetworkIndex && process.env.REACT_APP_FAUCET_PAGE) {
+    const holeskyIndex = NETWORKS_DATA.findIndex((networkData: any) => networkData.networkId === NETWORKS.HOLESKY);
+    return saveNetwork(holeskyIndex);
+  }
+  return saveNetwork(FIRST_NETWORK_INDEX);
+};
```

Comment:
This code became unreadable and hard to understand. 
Let's refactor it now and comment every code block to see what happens and why